app/controllers/lightweight_activities_controller.rb

```diff
@@ -80,20 +99,18 @@ def preview
   def single_page
     authorize! :read, @activity
     if !params[:run_key]
-      redirect_to activity_single_page_with_run_path(@activity, @run_key) and return
+      if @sequence
+        redirect_to sequence_activity_single_page_with_run_path(@sequence, @activity, @run_key) and return
+      else
+        redirect_to activity_single_page_with_run_path(@activity, @run_key) and return
+      end
```

Comment:
This also looks like it would lose the `mode=teacher-edition` url param. The old code looks like it would lose it as well.